app.rb

```diff
@@ -25,16 +26,31 @@ def initialize
   end
 
   get '/' do
-    erb :index, locals: { projects: key_list }
+    if authed?
+      erb :index, locals: { projects: key_list }
+    else
+      redirect '/auth/myusa'
+    end
+  end
+
+  get '/auth/myusa/callback' do
+    if env['omniauth.auth']
+      session[:authenticated] = true
```

Comment:
It is supposed to be secure, but I'll switch it out for `encrypted_cookie`, which is supposed to be more secure. Good call on the user_email bit, I'll add that.